react-forms/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,29 @@
+import { FormEvent } from 'react';
+import { useState } from 'react';
+
+export function RegistrationFormControlled() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>): void {
+    event.preventDefault();
+    const formData = new FormData(event.currentTarget);
```

Comment:
With a controlled form, there is no reason to use `FormData`. So you can just delete lines 10-13.